### PR TITLE
SQL-1094: add bounds checking for to_f32

### PR DESCRIPTION
**Pull request by terakilobyte**
Adding bounds checking for to_f32. I looked at the rest of the logic (in `to_f64()`), and it looks correct.

### CODE REVIEW

File: odbc/src/api/data.rs [hunk 1]
```diff
@@ -97,11 +99,13 @@ impl IntoCData for Bson {
...
 
     fn to_f32(&self) -> Result<f32> {
-        // this is mildly inefficient, unlike converting to i64 and then i32 which is essentially
-        // free (on x86_64 you can literally just change the register address from e.g., RAX to EAX,
-        // so it's literally a no-op), but since mongodb does not actually support 32 bit floats, we would need to do this
-        // conversion somewhere, anyway.
-        Ok(self.to_f64()? as f32)
+        let out = self.to_f64()?;
+        if out > f32::MAX as f64 || out < f32::MIN as f64 {
+            let _ = i64::MAX;
```
Comment (pmeredit) [inline odbc/src/api/data.rs:104]: What does this line do?

**Review by pmeredit — APPROVED** [review]
LGTM Other than I think that that `let _ = i64::MAX` doesn't do anything

File: odbc/src/api/data.rs (hunk 1 above)
Comment (terakilobyte) [reply odbc/src/api/data.rs]: Nothing. Not sure why I left that in there. Thanks for catching!

File: odbc/src/api/data.rs [hunk 2]
```diff
@@ -1487,6 +1555,32 @@ mod unit {
                 test_it!(bson, v);
             });
         }
+        #[test]
+        fn string_conversions_to_floats() {
+            type V = Vec<(&'static str, f64, Result<(), ()>, Option<&'static str>)>;
+            let strings: HashMap<String, V> = map! {
+                (-PI).to_string() => vec![
+                    ("f64", -PI, Ok(()), None),
+                    ("f32", -PI, Ok(()), Some(FRACTIONAL_TRUNCATION)),
```
Comment (nbagnard) [inline odbc/src/api/data.rs:1564]: I don't think it should (or will) return a Fractional truncation info. It looks like this was just overlooked and you should have None here, isn't it?
Same for PI string to f32 below.

**Review by nbagnard — APPROVED** [review]
LGTM!
There is only the string to float tests where I think there is a left over fractional truncation. The test will not check it, because the result is success if I get this right.

File: odbc/src/api/data.rs (hunk 2 above)
Comment (terakilobyte) [reply odbc/src/api/data.rs:1564]: It definitely won't. It looks like I need to update the test harness; this should have failed!

**Comment by terakilobyte** [comment]
evergreen merge

**Comment by terakilobyte** [comment]
evergreen merge

**Pull request merged**